device/cuda/src/utils/utils.cpp

```diff
@@ -0,0 +1,50 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Local include(s).
+#include "utils.hpp"
+
+#include "traccc/cuda/utils/definitions.hpp"
+
+namespace traccc::cuda::details {
+
+int get_device() {
+
+    int d = -1;
+    cudaGetDevice(&d);
+    return d;
```

Comment:
I'm not a fan of this.. If there is a CUDA error that should be immediately reported to the user by checking the result of `cudaGetDevice`.

Reply:
I really want to keep it like this.

This is what we went for in vecmem as well.

https://github.com/acts-project/vecmem/blob/main/cuda/src/utils/cuda_wrappers.cpp#L25

As it can help with making the code report a non-existence of CUDA devices happen without an error condition being raised.

Right now in this project if this function is called without any CUDA device, we're in trouble for sure. But later on this lack of an explicit check may still help us.